webapp/src/components/assetEditor/editor.tsx

```diff
@@ -159,7 +159,12 @@ export class AssetEditor extends Editor {
         fieldView.onHide(() => {
             const result = fieldView.getResult();
             if (asset.type == pxt.AssetType.Tilemap) result.data = this.updateTilemapTiles(result.data);
-            cb(result);
+
+            Promise.resolve(cb(result)).then(() => {
+                this.parent.saveBlocksToTypeScriptAsync().then((src) => {
+                    pkg.mainEditorPkg().setContentAsync("main.ts", src)
```

Comment:
Could we unify this with the normal blocks saving codepath somehow?

Reply:
it's a little weird because the default save path is very tied to the current editorFile, but in this case what we really want is to pretend the current file is main.blocks instead of assets.json. i could pass a file override to saveFileAsync/saveTypeScriptAsync? or if you have a better suggestion, totally welcome!

Reply:
also i'm gonna update so that we only do this save if we're editing a temporary asset

Reply:
ehhh, okay